### DISCUSSION TITLE
Should we introduce Prettier?

### DISCUSSION THREAD

**Discussion by maledr5** [justeattakeaway/pie#2122] [General]
I’ve noticed that some of my contributions to the code end up with formatting issues that are flagged during PR reviews. I think this might be happening because I use a different IDE than most of the team (IntelliJ), and I’m less familiar with the formatting conventions we use.

It’s a bit challenging to anticipate all the formatting rules, especially as we aim to make the codebase more accessible to contributions from people outside the team. This could lead to more instances of files being unintentionally reformatted.

I’d love to hear your thoughts on these questions:

Is there a way for my IDE to automatically apply our formatting rules that I might be missing? If so, would it make sense to document this for others who might use different tools?
Does anyone else see this as something we could improve?
If this resonates as a challenge, what do you think about introducing tools like Prettier to ensure consistent formatting across the codebase?

**Comment by ashleynolan** [comment]
So atm, we use ESLint for formatting our JS and Stylelint for our CSS.

These configs are setup as separate extensible plugins, that are used in the PIE (and Aperture) repos:
- [eslint-config-pie](https://github.com/justeattakeaway/pie/tree/main/packages/tools/eslint-config-pie)
- [stylelint-config-pie](https://github.com/justeattakeaway/pie/tree/main/packages/tools/stylelint-config-pie)

Usually, contributors will need to set-up a plugin within their IDE to hook into these configurations to do things like autofix on save.

I've started to write some linting docs this morning – I'll carry on with this on Monday, but may be good to add how to set this up on other editors: https://github.com/justeattakeaway/pie/wiki/Code-Linting-Tools
